### ISSUE TITLE
Cannot start fresh install of Immich because of permissions

### ISSUE THREAD

**Issue by christian-nils** [truenas/charts#2003]
Hello,

I hope this is the right place to post this. I am on Cobia 23.10. I am trying to install Immich from this catalog:

App version: 1.91.4
Chart Version: 3.0.2

I have set Immich storage to:
`/mnt/app-pool/apps/immich/data`

I have set Immich backup to
`/mnt/app-pool/apps/immich/backup`

They are both owned by the user and the group `apps` but the postgres pod fails with:

```shell
2024-01-06 11:00:37.843646+01:00chmod: changing permissions of '/var/lib/postgresql/data': Operation not permitted
2024-01-06 11:00:37.876615+01:00The files belonging to this database system will be owned by user "postgres".
2024-01-06 11:00:37.876669+01:00This user must also own the server process.
2024-01-06 11:00:37.876680+01:002024-01-06T11:00:37.876680933+01:00
2024-01-06 11:00:37.876691+01:00The database cluster will be initialized with locale "en_US.utf8".
2024-01-06 11:00:37.876700+01:00The default database encoding has accordingly been set to "UTF8".
2024-01-06 11:00:37.876723+01:00The default text search configuration will be set to "english".
2024-01-06 11:00:37.876733+01:002024-01-06T11:00:37.876733567+01:00
2024-01-06 11:00:37.876743+01:00Data page checksums are disabled.
2024-01-06 11:00:37.876752+01:002024-01-06T11:00:37.876752108+01:00
2024-01-06 11:00:37.876777+01:00initdb: error: could not change permissions of directory "/var/lib/postgresql/data": Operation not permitted
2024-01-06 11:00:37.877963+01:00fixing permissions on existing directory /var/lib/postgresql/data ... 
```

I am not sure why, I did install Immich in the past and I did not have that issue, I wonder if something is wrong in my setup. Could you please let me know what I can try to fix this ?

Thanks in advance

**Comment by stavros-k** [comment]
Hello, postgres must be user and group of `999` (this is the id that is mapped to the `postgres` user inside the container

**Comment by christian-nils** [comment]
Thank you very much for your quick answer. I will check later today and report back. Any pointers how I could have find this information without bothering you?

**Comment by stavros-k** [comment]
It's usually in both Readme.md and metadata.yaml in the app
https://github.com/truenas/charts/blob/master/library/ix-dev/community/immich/metadata.yaml
https://github.com/truenas/charts/blob/master/library/ix-dev/community/immich/README.md

But looking at it now, seems that this one was missed the update.
I plan to "revisit" those and make sure are updated soon-ish

**Comment by stavros-k** [comment]
But thinking about it, it was supposed to prepare the database directory permissions on the first installation.

Did you have ACLs or other permissions set on the database hostPath(s) prior to the installation?

**Comment by christian-nils** [comment]
So the postgres storage hostpath and its backup hostpath were datasets where with the owner and group were set to `apps` (I had added write permission to group also). Now that you mentioned the permission setup on the first installation, it actually failed if I recall correctly which could explain the messages I have in the original post.

I have deleted the datasets then I created them, set the owner and group to `apps` and recreated the app, it just worked. The permissions were changed (owner=netdata, group=docker) and the app was now accessible. So I am not sure what I missed before.

![image](https://github.com/truenas/charts/assets/2658014/6c3c75a5-5949-4c87-889d-c3b524ff3d9b)

Thanks again for the help :) and sorry for the troubles.
